lib/mantav2-migrate/do_snaplink_cleanup.js

```diff
@@ -0,0 +1,729 @@
+                                    'Created delink scripts in %s/delink/',
+                                    DBDIR
+                                );
+                                ui.info('  stordelink scripts:');
+                                for (let script of scripts.stordelink) {
+                                    ui.info('    ' + path.basename(script));
+                                }
+                                ui.info('  moraydelink scripts:');
+                                for (let script of scripts.moraydelink) {
+                                    ui.info('    ' + path.basename(script));
+                                }
+                            }
+                            next();
+                        }
+                    );
+                },
+
+                function confirmDelinkScriptsHaveBeenRun(ctx, next) {
+                    ui.info(
+                        [
+                            '',
+                            '',
+                            '# Phase 4: Running delink scripts',
+                            '',
+                            '"Delink" scripts have been generated from the snaplink listings',
+                            'from the previous phase. In this phase, you must:',
+                            '',
+                            '1. Copy each "*_stordelink.sh" script to the appropriate storage',
+                            '   node and run it there. There are %d to run:',
+                            '',
+                            '        # {storage_id}_stordelink.sh',
+                            '        ls %s/delink/*_stordelink.sh',
+                            '',
+                            '   Use the following to help locate each storage node:',
+                            '',
+                            '        manta-adm show -a -o service,storage_id,datacenter,zonename,gz_host,gz_admin_ip | grep ^storage',
+                            '',
+                            '2. **Only after** all stordelink scripts have been run, copy',
+                            '   each "*_moraydelink.sh" script to a moray zone for the',
+                            '   appropriate shard and run it there. There is one script',
+                            '   script for each Manta index shard (%d):',
+                            '',
+                            '        # {shard}_moraydelink.sh',
+                            '        ls %s/delink/*_moraydelink.sh',
+                            '',
+                            '   Use the following to help locate a moray for each shard:',
+                            '',
+                            '        manta-adm show -o service,shard,zonename,gz_host,gz_admin_ip | grep ^moray',
+                            '',
+                            'When you are sure you have run all these scripts, then answer',
+                            'the following to proceed. *WARNING* Be sure you have run all',
```

Comment:
Can we run another metadata scan after the scripts are run to verify that they have been successful? Would that massively slow down the cleanup process, or is it a useful safety net?

Reply:
We *could*, but yes it would massively slow down the cleanup process. It would require another run of the sherlock script on each index shard. We expect those to take hours each in SPC.